plugins/outputs/http/http.go

```diff
@@ -0,0 +1,196 @@
+func (h *Http) Connect() error {
+	h.client = http.Client{
+		Transport: &http.Transport{
+			Dial: (&net.Dialer{
+				Timeout: time.Duration(h.DialTimeOut) * time.Second,
+			}).Dial,
+			ResponseHeaderTimeout: time.Duration(h.ResHeaderTimeout) * time.Second,
+		},
+	}
+
+	return nil
+}
+
+// Close is not implemented. Because http.Client not provided connection close policy. Instead, uses the response.Body.Close() pattern.
+func (h *Http) Close() error {
+	return nil
+}
+
+// Description A plugin that can transmit metrics over HTTP
+func (h *Http) Description() string {
+	return "A plugin that can transmit metrics over HTTP"
+}
+
+// SampleConfig provides sample example for developer
+func (h *Http) SampleConfig() string {
+	return sampleConfig
+}
+
+// Writes metrics over HTTP POST
+func (h *Http) Write(metrics []telegraf.Metric) error {
+	var mCount int
+	var reqBodyBuf []byte
+
+	for _, metric := range metrics {
+		buf, err := h.serializer.Serialize(metric)
+
+		if err != nil {
+			return fmt.Errorf("E! Error serializing some metrics: %s", err.Error())
+		}
+
+		reqBodyBuf = append(reqBodyBuf, buf...)
+		mCount++
+	}
+
+	reqBody, err := makeReqBody(h.serializer, reqBodyBuf, mCount)
+
+	if err != nil {
+		return fmt.Errorf("E! Error serialized metric is not assembled : %s", err.Error())
+	}
+
+	if err := h.write(reqBody); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (h *Http) write(reqBody []byte) error {
+	req, err := http.NewRequest(POST, h.URL, bytes.NewBuffer(reqBody))
+
+	for _, httpHeader := range h.HttpHeaders {
+		keyAndValue := strings.Split(httpHeader, ":")
+		req.Header.Set(keyAndValue[0], keyAndValue[1])
+	}
+
+	req.Close = true
+
+	res, err := h.client.Do(req)
+
+	if err := h.isOk(res, err); err != nil {
+		return err
+	}
+
+	res.Body.Close()
+
+	return err
+}
+
+func (h *Http) isOk(res *http.Response, err error) error {
+	if res == nil || err != nil {
+		return fmt.Errorf("E! %s request failed! %s.", h.URL, err.Error())
+	}
+
+	if !h.isExpStatusCode(res.StatusCode) {
+		return fmt.Errorf("E! %s response is unexpected status code : %d.", h.URL, res.StatusCode)
+	}
+
+	return nil
+}
+
+func (h *Http) isExpStatusCode(resStatusCode int) bool {
+	if h.expStatusCode == nil {
+		h.expStatusCode = make(map[int]bool)
+
+		for _, expectedStatusCode := range h.ExpStatusCodes {
+			h.expStatusCode[expectedStatusCode] = true
+		}
+	}
+
+	if h.expStatusCode[resStatusCode] {
+		return true
+	}
+
+	return false
+}
+
+// makeReqBody translates each serializer's converted metric into a request body.
+func makeReqBody(serializer serializers.Serializer, reqBodyBuf []byte, mCount int) ([]byte, error) {
+	switch serializer.(type) {
+	case *json.JsonSerializer:
+		var arrayJsonObj []byte
+		arrayJsonObj = append(arrayJsonObj, []byte("[")...)
+		arrayJsonObj = append(arrayJsonObj, reqBodyBuf...)
+		arrayJsonObj = append(arrayJsonObj, []byte("]")...)
+		return bytes.Replace(arrayJsonObj, []byte("\n"), []byte(","), mCount-1), nil
```

Comment:
I understand why you did this since the Serializer interface does not provide a way to process a list of metrics, but I don't think it is robust enough.  I will update the Serializer interface to take a `[]telegraf.Metrics` and then we can do this in a better way.

Reply:
You can see 92 line about serializer.